client/signup/config/flows.js

```diff
@@ -190,6 +191,20 @@ function removeUserStepFromFlow( flow ) {
 	} );
 }
 
+function filterDesignTypeInFlow( flow ) {
+	if ( ! flow ) {
+		return;
+	}
+
+	if ( ! includes( flow.steps, 'design-type' ) || 'designTypeWithStore' !== abtest( 'signupStore' ) ) {
+		return flow;
+	}
+
+	return assign( {}, flow, {
+		steps: flow.steps.splice( flow.steps.indexOf( 'design-type' ), 1, 'design-type-with-store' )
```

Comment:
This `Object.assign` implies that you don't want to mutate `flow.steps`, which I think is probably a good idea. However,  `flow.steps.splice` will change the original array and only returns the steps that have been changed. Check this out:

```js
var foos = ['one', 'two', 'three'];
foos.splice(foos.indexOf('three'), 1, 'four');
["three"]
foos
["one", "two", "four"]
foos = ['one', 'two', 'three'];
foos.map( foo => foo === 'three' ? 'four' : foo )
["one", "two", "four"]
foos
["one", "two", "three"]
```